src/util/include/util/Assert.h

```diff
@@ -0,0 +1,27 @@
+/*
+ * Xournal++
+ *
+ * assert
+ *
+ * @author Xournal++ Team
+ * https://github.com/xournalpp/xournalpp
+ *
+ * @license GNU GPLv2 or later
+ */
+#pragma once
+
+#include <string>
+
+#ifdef NDEBUG
+#define xoj_assert(expr) static_cast<void>(0)
+#define xoj_assert(expr, msg) static_cast<void>(0)
+#else
+namespace xoj::util {
+void assertFailure(const char* expr, const std::string& msg, const char* fileName, int line, const char* funcName);
```

Comment:
Either inline the function into the header, or mark it as `[[noreturn]]` since the compiler should know, that the program terminates here. (std::abort also has the attribute, but other compilation units can't know if it is not inlined / marked explicit)



```suggestion
[[noreturn]] void assertFailure(const char* expr, const std::string& msg, const char* fileName, int line, const char* funcName);
```